freezegun/api.py

```diff
@@ -35,6 +38,60 @@
     import copyreg
 
 
+# keep a cache of module attributes otherwise freezegun will need to analyze too many modules all the time
+_GLOBAL_MODULES_CACHE = None
+
+
+def _get_global_modules_cache():
+    global _GLOBAL_MODULES_CACHE
+    if _GLOBAL_MODULES_CACHE is None:
+        _GLOBAL_MODULES_CACHE = _setup_modules_cache()
+    return _GLOBAL_MODULES_CACHE
+
+
+def _get_module_attributes(module):
+    result = []
+    for attribute_name in dir(module):
+        try:
+            attribute_value = getattr(module, attribute_name)
+        except (ImportError, AttributeError, TypeError):
+            # For certain libraries, this can result in ImportError(_winreg) or AttributeError (celery)
+            continue
+        else:
+            result.append((attribute_name, attribute_value))
+    return result
+
+
+def _setup_modules_cache():
+    # FIXME: move this definition to be at the top-level
+    real_time_object_ids = set(id(obj) for obj in real_date_objects)
+    result = {}
+    for mod_name, module in list(sys.modules.items()):
+        # ignore this module
+        if mod_name == __name__:
```

Comment:
@spulec: in the `_freeze_time.start` method I saw a similar logic (`if mod_name is None or module is None`) but I didn't understand the `None` checks and didn't put them here. When are they `None`?